airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/EnrollmentButton/EnrollmentButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button, ButtonVariant } from "components/ui/Button";
+import { FlexContainer, FlexItem } from "components/ui/Flex";
+
+import { useModalService } from "hooks/services/Modal";
+import { useStripeCheckout } from "packages/cloud/services/stripe/StripeService";
+import { useCurrentWorkspaceId } from "services/workspaces/WorkspacesService";
+
+import { EnrollmentModalContent } from "./EnrollmentModal";
+
+interface FCPEnrollmentButtonProps {
+  buttonTextKey?: string;
+  variant?: ButtonVariant;
+}
+
+export const FCPEnrollmentButton: React.FC<FCPEnrollmentButtonProps> = ({
+  buttonTextKey = "freeConnectorProgram.enrollmentModal.enrollButtonText",
+  variant = "primary",
+}) => {
+  const { openModal, closeModal } = useModalService();
+  const { mutateAsync: createCheckout } = useStripeCheckout();
+  const workspaceId = useCurrentWorkspaceId();
+
+  const showEnrollmentModal = () => {
```

Comment:
We'll need to call this from a span-acting-as-a-button in the inline banner ([here](https://github.com/airbytehq/airbyte/pull/21315)).  Can we split out this logic so it's reusable?

Reply:
Good point, I've refactored this into the `useShowEnrollmentModal()` hook which can then be used anywhere. I got rid of the `EnrollmentButton` component, since it became pretty redundant!